diff --git a/drivers/usb/musb-new/musb_core.c b/drivers/usb/musb-new/musb_core.c
index add6b6c3..e7798d32 100644
--- a/drivers/usb/musb-new/musb_core.c
+++ b/drivers/usb/musb-new/musb_core.c
@@ -922,7 +922,6 @@ int musb_start(struct musb *musb)
 
 	/* put into basic highspeed mode and start session */
 	musb_writeb(regs, MUSB_POWER, MUSB_POWER_ISOUPDATE
-						| MUSB_POWER_HSENAB
 						/* ENSUSPEND wedges tusb */
 						/* | MUSB_POWER_ENSUSPEND */
 						);
diff --git a/drivers/usb/musb-new/musb_gadget.c b/drivers/usb/musb-new/musb_gadget.c
index 8ba98d8c..359622da 100644
--- a/drivers/usb/musb-new/musb_gadget.c
+++ b/drivers/usb/musb-new/musb_gadget.c
@@ -1680,6 +1680,7 @@ static int musb_gadget_wakeup(struct usb_gadget *gadget)
 
 	power = musb_readb(mregs, MUSB_POWER);
 	power |= MUSB_POWER_RESUME;
+	power &= ~MUSB_POWER_HSENAB;
 	musb_writeb(mregs, MUSB_POWER, power);
 	dev_dbg(musb->controller, "issue wakeup\n");
 
@@ -1688,6 +1689,7 @@ static int musb_gadget_wakeup(struct usb_gadget *gadget)
 
 	power = musb_readb(mregs, MUSB_POWER);
 	power &= ~MUSB_POWER_RESUME;
+	power &= ~MUSB_POWER_HSENAB;
 	musb_writeb(mregs, MUSB_POWER, power);
 done:
 	spin_unlock_irqrestore(&musb->lock, flags);
